staging/src/k8s.io/sample-apiserver/README.md

```diff
@@ -24,6 +24,59 @@ HEAD of this repo will match HEAD of k8s.io/apiserver, k8s.io/apimachinery, and
 `sample-apiserver` is synced from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver.
 Code changes are made in that location, merged into `k8s.io/kubernetes` and later synced here.
 
+## Normal Build and Deploy
+
+### Changes to the Types
+
+If you change the API object type definitions in any of the
+`pkg/apis/.../types.go` files then you will need to update the files
+generated from the type definitions.  There is no convenient script to
+do this in the context of `$GOPATH/src/k8s.io/sample-apiserver`.
+However, if you have fetched the whole `k8s.io/kubernetes` repository
+then there _is_ a convenient script that you can use.  With
+`$GOPATH/src/k8s.io/kubernetes/staging/src/k8s.io/sample-apiserver` as
+your current working directory, invoke `hack/update-codegen.sh`; the
+script takes no arguments.
+
+### Build the Binary
+
+With `sample-apiserver` as your current working directory, issue the
+following command:
+
+```
+CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o artifacts/simple-image/kube-sample-apiserver
+```
+
+### Build the Container Image
+
+With `sample-apiserver` as your current working directory, issue the
+following commands with `MYPREFIX` and `MYTAG` replaced by something
+suitable.
+
+```
+docker build -t MYPREFIX/kube-sample-apiserver:MYTAG ./artifacts/simple-image
+docker push MYPREFIX/kube-sample-apiserver:MYTAG
+```
+
+
+### Build Binary and Container Image from Current Master Source
+
+There is a Dockerfile at
+https://github.com/kubernetes/kubernetes/blob/master/test/images/sample-apiserver/Dockerfile
```

Comment:
This should point to the github.com/kubernetes/sample-apiserver repo, not to k/k.

Reply:
The Dockerfile that builds the binary is not in the github.com/kubernetes/sample-apiserver repo.

Reply:
But it should.

Reply:
Are you suggesting to _move_ it from k/k/test/images into the staging area, or _copy_ it, or ...?

Reply:
That Dockerfile looks seriously old to me.  It uses go 1.10 to build release 1.10.  It does not deal with the added complexities involved in working with the latest code.  I think the current contents of that Dockerfile are no longer a good guide.  I am updating this repo's README to say what works to build the latest code.